### ISSUE TITLE
Provide a means to customize the FOV constraint for Vision Sources

### ISSUE THREAD

**Issue by caewok** [foundryvtt/foundryvtt#7271]
In 10.270, `VisionSource.prototype.initialize` now sets the `fov` by constraining the line-of-sight polygon with a circle. I think it would be useful to allow systems or modules to customize this constraint. Could you add either a method or a property that would allow modules or systems to re-define the `fovCircle` variable in `initialize` to some other object recognized by `applyConstraint`? (E.g., any PIXI.Rectangle or PIXI.Polygon would probably work here.) 

### User Experience

While a circle seems like the obvious choice, I am sure users can come up with reasons why they might prefer some other shape to define the field of vision. For example, using a square to match the square grid or a hexagon to match the hex grid. This change would allow a module or system to easily re-define the shape without overwriting the `initialize` method. 

### Priority/Importance

Not too important, but I think this is an easy change and might help with developing specialized vision modes.

**Comment by aaclayton** [comment]
Makes sense as an approach to API improvement, although I don't know how a module would make this change without patching the `VisionSource` prototype. Maybe this is good enough.

**Comment by pandanielxd** [comment]
Hey, I was wondering whether this was still being considered for a future Foundry version? I really like the idea of square vision at my table if it is ever in the cards.

**Comment by dev7355608** [comment]
Related issue: #10795.
